### PR TITLE
Dispatch domain events in CategoryBundle

### PR DESCRIPTION
**Pull request by niklasnatter**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Related issues/PRs | builds upon #5840
| License | MIT

#### What's in this PR?

This PR adjusts the `CategoryManager` to dispatch domain events.

### CODE REVIEW

File: src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryCreatedEvent.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    public function getEventPayload(): array
+    {
+        return $this->payload;
+    }
+
+    public function getResourceKey(): string
+    {
+        return 'categories';
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryCreatedEvent.php:64]: define a constant for this

File: src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryCreatedEvent.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    public function getResourceLocale(): ?string
+    {
+        return $this->locale;
+    }
+
+    public function getResourceTitle(): ?string
+    {
+        return $this->category->findTranslationByLocale($this->locale)->getTranslation();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryCreatedEvent.php]: `$this->category->findTranslationByLocale($this->locale)` could return `null`

File: src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryMovedEvent.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    public function getResourceId(): string
+    {
+        return (string) $this->category->getId();
+    }
+
+    public function getResourceTitle(): ?string
+    {
+        return $this->category->findTranslationByLocale($this->category->getDefaultLocale())->getTranslation();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryMovedEvent.php]: same above the `findTranslationByLocale` can return null would more be a race condition  but maybe we should still not fail hard if it happens

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryMovedEvent.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    public function getEventType(): string
+    {
+        return 'moved';
+    }
+
+    public function getEventContext(): array
+    {
+        return [
+            'previousParentId' => $this->previousParentId,
+            'newParentId' => $this->category->getParent()->getId(),
```
Comment (niklasnatter) [inline src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryMovedEvent.php]: i suspect that `$this->category->getParent()` could return null

File: src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php [hunk 5]
```diff
@@ -272,7 +290,10 @@ public function move($id, $parent)
             throw new CategoryIdNotFoundException($parent);
         }
 
+        $previousParentId = $category->getParent()->getId();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php]: @nnatter think this `getParent()` call could also be `null` when it is a root Category moved into a child

**Review by alexander-schranz — APPROVED** [review]

**Review by wachterjohannes — APPROVED** [review]

**Pull request merged**